src/dns_private.tf

```
@@ -22,3 +22,22 @@ resource "azurerm_private_dns_a_record" "private_dns_a_record_db_nodo" {
   ttl                 = 60
   records             = var.dns_a_reconds_dbnodo_ips
 }
+
+resource "azurerm_private_dns_zone" "redis_pagopa_proxy_dns_zone" {
```

```suggestion
resource "azurerm_private_dns_zone" "privatelink_redis_cache_windows_net" {
```
private dns zone and virtual network links names should be generic: we can have only 1 dns zone and 1 virtual network link for vnet. So if u have 2 redis cache u have to reuse this dns zone and vnet link.
I suggest to use `privatelink_redis_cache_windows_net` and `privatelink_redis_cache_windows_net_vnet` as here https://github.com/pagopa/selfcare-infra/blob/main/src/core/dns_private.tf#L56

---

What about also renaming `pagopa_proxy_redis_private_endpoint_enabled` to `redis_private_endpoint_enabled`?

If understood everything correctly the virtual network link and the dns zone can be used by many different redis instances (as long as they are under the same vnet), so it should make sense to generify the variable enabling those too.

---

it's correct we will reuse the redis dns zone and vnet link in future redis instances. Now we have two ways:
1. use a single variable `redis_private_endpoint_enabled`
2. set a count condition with OR, for example

```
count = (var.pagopa_proxy_redis_private_endpoint_enabled || var.service_b_redis_private_endpoint_enabled || var.service_c_redis_private_endpoint_enabled) ? 1 : 0
```

PS: we can remove pagopa_proxy_enabled from dns zone and vnet link count condition